### ISSUE TITLE
Add snippets for all Symbol classes

### ISSUE THREAD

**Issue by hhkaos** [Esri/arcgis-js-vscode-snippets#33]
Today the extension contain just a few snippets 

Snippet|Description
-- | --
|simpleMarkerSymbol | Creates a SimpleMarkerSymbol in a MapView. Contains placeholder for style.
|pictureMarkerSymbol | Creates a PictureMarkerSymbol in a MapView.
|simpleLineSymbol | Creates a SimpleLineSymbol in a MapView. Contains placeholder for style, cap and join.
|simpleFillSymbol | Creates a SimpleFillSymbol in a MapView. Contains placeholder for style.
|pictureFillSymbol | Creates a PictureFillSymbol in a MapView.

But this is just a small subset of all the [Symbols]([Symbols](https://developers.arcgis.com/javascript/latest/visualization/symbols-color-ramps/symbols/)).

It would be nice to have snippets for the following classes: (*just the ones in bold; 21 in total*)

- **[CIMSymbol](https://developers.arcgis.com/javascript/latest/api-reference/esri-symbols-CIMSymbol.html)**
- **[WebStyleSymbol](https://developers.arcgis.com/javascript/latest/api-reference/esri-symbols-WebStyleSymbol.html)**. Think that adding the enums for [WebStyleSymbol2D](https://developers.arcgis.com/javascript/latest/visualization/symbols-color-ramps/esri-web-style-symbols-2d/) and the [WebStyleSymbol3D](https://developers.arcgis.com/javascript/latest/visualization/symbols-color-ramps/esri-web-style-symbols-3d/) is too much
- For [2D Symbols](https://developers.arcgis.com/javascript/latest/api-reference/esri-symbols-Symbol.html)
	- [MarkerSymbol](https://developers.arcgis.com/javascript/latest/api-reference/esri-symbols-MarkerSymbol.html) (Points)
		- [**PictureMarkerSymbol**](https://developers.arcgis.com/javascript/latest/api-reference/esri-symbols-PictureMarkerSymbol.html)
		- [**SimpleMarkerSymbol**](https://developers.arcgis.com/javascript/latest/api-reference/esri-symbols-SimpleMarkerSymbol.html)
	- [LineSymbol](https://developers.arcgis.com/javascript/latest/api-reference/esri-symbols-LineSymbol.html) (Polylines)
		- [**SimpleLineSymbol**](https://developers.arcgis.com/javascript/latest/api-reference/esri-symbols-LineSymbol.html)
	- [FillSymbol](https://developers.arcgis.com/javascript/latest/api-reference/esri-symbols-FillSymbol.html) (Polygons) 
		- [**PictureFillSymbol**](https://developers.arcgis.com/javascript/latest/api-reference/esri-symbols-PictureFillSymbol.html)
		- [**SimpleFillSymbol**](https://developers.arcgis.com/javascript/latest/api-reference/esri-symbols-SimpleFillSymbol.html)
	- [**Text Symbol**](https://developers.arcgis.com/javascript/latest/api-reference/esri-symbols-TextSymbol.html) (Labels)
- For [3D symbols](https://developers.arcgis.com/javascript/latest/api-reference/esri-symbols-Symbol3D.html):
	- [PointSymbol3D](https://developers.arcgis.com/javascript/latest/api-reference/esri-symbols-PointSymbol3D.html) 
		 - **IconSymbol3DLayer**
		 - **ObjectSymbol3DLayer**
		 - **TextSymbol3DLayer**
	- [LineSymbol3D](https://developers.arcgis.com/javascript/latest/api-reference/esri-symbols-LineSymbol3D.html)
		 - **LineSymbol3DLayer**
		 - **PathSymbol3DLayer**
	- [PolygonSymbol3D](https://developers.arcgis.com/javascript/latest/api-reference/esri-symbols-PolygonSymbol3D.html) 
		- **FillSymbol3DLayer**
		- **ExtrudeSymbol3DLayer**
		- **WaterSymbol3DLayer**
		- **LineSymbol3DLayer**
		- **IconSymbol3DLayer**
		- **ObjectSymbol3DLayer**
		- **TextSymbol3DLayer**
	- [MeshSymbol3D](https://developers.arcgis.com/javascript/latest/api-reference/esri-symbols-MeshSymbol3D.html) 
		- **FillSymbol3DLayer**

Instead of just the properties:
```js
{
  type: "simple-marker",
  color: [255, 255, 255, 0.25],
  size: 12,
  style: "circle",
  outline: {
    width: 1,
    color: [255, 255, 255, 1]
  }
}

/* Snippet
 "SimpleMarkerSymbol": {
    "body": [
      "{",
      "\ttype: \"simple-marker\",",
      "\tcolor: [255, 255, 255, 0.25],",
      "\tsize: 12,",
      "\tstyle: \"${1|circle,square,cross,x,diamond,triangle,path|}\",",
      "\toutline: {",
      "\t\twidth: 1,",
      "\t\tcolor: [255, 255, 255, 1]",
      "\t}",
      "}"
    ],
    "description": "Create a SimpleMarkerSymbol",
    "prefix": "simpleMarkerSymbol"
  }
*/
```

I would suggest something like:

```js
new SimpleMarkerSymbol({
  type: "simple-marker", // required only when autocasting
  color: [255, 255, 255, 0.25],
  size: 12,
  style: "circle",
  outline: {
    width: 1,
    color: [255, 255, 255, 1]
  }
})

/* Snippet
"": {
"prefix": "",
"body": [
	"new SimpleMarkerSymbol({",
	"\ttype: \"simple-marker\", // required only when autocasting",
	"\tcolor: [255, 255, 255, 0.25],",
	"\tsize: 12,",
	"\tstyle: \"${1|circle,square,cross,x,diamond,triangle,path|}\",",
	"\toutline: {",
	"\t\twidth: 1,",
	"\t\tcolor: [255, 255, 255, 1]",
	"\t}",
	"})",
],
"description": ""
}
*/
```

Does it make sense @kellyhutchins ? // cc: @RalucaNicola

**Comment by kellyhutchins** [comment]
Yes 100% it would be fantastic to have snippets for all of the symbol classes

**Comment by sthiyaku** [comment]
@hhkaos can you please assign this to me ?

**Comment by hhkaos** [comment]
Sure, done! 👍 😄
